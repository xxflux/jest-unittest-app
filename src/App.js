import React, { Component } from 'react';

class App extends Component {
  constructor() {
    super();
    this.state = {
      count: 0,
    }
  }

  makeIncrease = amount => () =>
    this.setState(prevState => ({
      count: prevState.count + amount,
    }));

  increase = this.makeIncrease(1);

  render() {
    return (
      <div>
        <h1>Click it, and count it!</h1>
        <h3>Count: {this.state.count}</h3>
        <button className="increase" onClick={this.increase}>Click me to count</button>
      </div>
    )
  }
}

export default App;
