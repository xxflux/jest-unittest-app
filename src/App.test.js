import renderer from 'react-test-renderer';
import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('Click it Count it starts', () => {
  it('starts with 0', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('h3').text();
    expect(text).toEqual('Count: 0');
  });

  it('click it, increase it by 1', () => {
    const wrapper = shallow(<App />);
    const increaseBtn = wrapper.find('button.increase');
    increaseBtn.simulate('click');
    const text = wrapper.find('h3').text();
    expect(text).toEqual('Count: 1');
  });

  it('check the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
