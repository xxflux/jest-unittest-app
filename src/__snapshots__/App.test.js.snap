// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click it Count it starts check the snapshot 1`] = `
<div>
  <h1>
    Click it, and count it!
  </h1>
  <h3>
    Count: 
    0
  </h3>
  <button
    className="increase"
    onClick={[Function]}
  >
    Click me to count
  </button>
</div>
`;
